### ISSUE TITLE
CMake tries checking sys/time.h for sndfile and libsoundio

### ISSUE THREAD

**Issue by knockshore** [LMMS/lmms#2815]
![image](https://cloud.githubusercontent.com/assets/7346503/15734502/0e2e021c-28ae-11e6-9e15-191aefce21c2.png)

I checked the CMakeLists.txt, I'm not sure this block is working. I had to set the `WANT_SOUNDIO` to `OFF` at the top to make it work.
![image](https://cloud.githubusercontent.com/assets/7346503/15734527/65e2bba6-28ae-11e6-8666-52959e2b21cc.png)

**Configure log:**

```
$ ../cmake/build_mingw64.sh

Custom cmake vars: (blank = system default)
-----------------------------------------
* CMAKE_C_COMPILER                     : /mingw64/bin/gcc
* CMAKE_CXX_COMPILER                   : /mingw64/bin/g++
* CMAKE_RC_COMPILER                    : /mingw64/bin/windres
* WINDRES                              : /mingw64/bin/windres
* ENV{PKG_CONFIG}                      :
* MINGW_TOOL_PREFIX32                  :
* CMAKE_C_COMPILER32                   : /mingw32/bin/gcc
* CMAKE_CXX_COMPILER32                 : /mingw32/bin/g++
* STRIP                                : /mingw64/bin/strip
* QT_BINARY_DIR                        : /mingw64/bin
* QT_QMAKE_EXECUTABLE                  : /mingw64/bin/qmake

-- The C compiler identification is GNU 5.3.0
-- The CXX compiler identification is GNU 5.3.0
-- Check for working C compiler: /mingw64/bin/gcc
-- Check for working C compiler: /mingw64/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /mingw64/bin/g++
-- Check for working CXX compiler: /mingw64/bin/g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /opt/mingw64/bin/x86_64-w64-mingw32-pkg-config
-- Found Git: /usr/bin/git.exe (found version "2.8.2")
Git not found.  Using release info from /CMakeLists.txt

Configuring LMMS
--------------------------
* Build version               : 1.1.91
*   Major version             : 1
*   Minor version             : 1
*   Patch version             : 91
*   Suffix version            :
*

Optional Version Usage:
--------------------------
*   Override version:           -DFORCE_VERSION=x.x.x-x
*   Disable hash suffix:        -DFORCE_VERSION=internal

PROCESSOR: x86_64
-- Target host is 64 bit
-- Looking for include file stdint.h
-- Looking for include file stdint.h - found
-- Looking for include file stdlib.h
-- Looking for include file stdlib.h - found
-- Looking for include file pthread.h
-- Looking for include file pthread.h - found
-- Looking for include file semaphore.h
-- Looking for include file semaphore.h - found
-- Looking for include file unistd.h
-- Looking for include file unistd.h - found
-- Looking for include file sys/types.h
-- Looking for include file sys/types.h - found
-- Looking for include file sys/ipc.h
-- Looking for include file sys/ipc.h - not found
-- Looking for include file sys/shm.h
-- Looking for include file sys/shm.h - not found
-- Looking for include file sys/time.h
-- Looking for include file sys/time.h - found
-- Looking for include file sched.h
-- Looking for include file sched.h - found
-- Looking for include file sys/soundcard.h
-- Looking for include file sys/soundcard.h - not found
-- Looking for include file soundcard.h
-- Looking for include file soundcard.h - not found
-- Looking for include file fcntl.h
-- Looking for include file fcntl.h - found
-- Looking for include file sys/ioctl.h
-- Looking for include file sys/ioctl.h - not found
-- Looking for include file ctype.h
-- Looking for include file ctype.h - found
-- Looking for include file string.h
-- Looking for include file string.h - found
-- Looking for include file process.h
-- Looking for include file process.h - found
-- Looking for include file locale.h
-- Looking for include file locale.h - found
-- Looking for Q_WS_X11
-- Looking for Q_WS_X11 - not found
-- Looking for Q_WS_WIN
-- Looking for Q_WS_WIN - found
-- Looking for Q_WS_QWS
-- Looking for Q_WS_QWS - not found
-- Looking for Q_WS_MAC
-- Looking for Q_WS_MAC - not found
-- Found Qt4: /mingw64/bin/qmake (found suitable version "4.8.7", minimum required is "4.6.0;COMPONENTS;QtCore;QtGui;QtXml")
-- Found Qt translations in /mingw64/share/qt4/translations/
-- Checking for module 'sndfile>=1.0.11'
--
CMake Error at /usr/share/cmake-3.5.2/Modules/FindPkgConfig.cmake:363 (message):
  A required package was not found
Call Stack (most recent call first):
  /usr/share/cmake-3.5.2/Modules/FindPkgConfig.cmake:528 (_pkg_check_modules_internal)
  CMakeLists.txt:178 (PKG_CHECK_MODULES)


-- Configuring incomplete, errors occurred!
See also "/home/karti/lmms/build/CMakeFiles/CMakeOutput.log".
See also "/home/karti/lmms/build/CMakeFiles/CMakeError.log".

```

Platform:
Windows 10 64-bit
MINGW64_NT-10.0 [NA] 2.5.1(0.297/5/3) 2016-05-16 10:51 x86_64 Msys

Edit:
**CMake Error log:**

```
Determining if files sys/ipc.h exist failed with the following output:
Change Dir: /home/karti/lmms/build/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/make.exe" "cmTC_75fce/fast"
/usr/bin/make -f CMakeFiles/cmTC_75fce.dir/build.make CMakeFiles/cmTC_75fce.dir/build
make[1]: Entering directory '/home/karti/lmms/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_75fce.dir/CheckIncludeFiles.c.obj
/mingw64/bin/gcc   @CMakeFiles/cmTC_75fce.dir/includes_C.rsp  -o CMakeFiles/cmTC_75fce.dir/CheckIncludeFiles.c.obj   -c /home/karti/lmms/build/CMakeFiles/CMakeTmp/CheckIncludeFiles.c
C:/msys64/home/karti/lmms/build/CMakeFiles/CMakeTmp/CheckIncludeFiles.c:2:21: fatal error: sys/ipc.h: No such file or directory
compilation terminated.
make[1]: *** [CMakeFiles/cmTC_75fce.dir/build.make:67: CMakeFiles/cmTC_75fce.dir/CheckIncludeFiles.c.obj] Error 1
make[1]: Leaving directory '/home/karti/lmms/build/CMakeFiles/CMakeTmp'
make: *** [Makefile:126: cmTC_75fce/fast] Error 2

Source:
/* */
#include <sys/ipc.h>


int main(void){return 0;}

Determining if files sys/shm.h exist failed with the following output:
Change Dir: /home/karti/lmms/build/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/make.exe" "cmTC_c4e7a/fast"
/usr/bin/make -f CMakeFiles/cmTC_c4e7a.dir/build.make CMakeFiles/cmTC_c4e7a.dir/build
make[1]: Entering directory '/home/karti/lmms/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_c4e7a.dir/CheckIncludeFiles.c.obj
/mingw64/bin/gcc   @CMakeFiles/cmTC_c4e7a.dir/includes_C.rsp  -o CMakeFiles/cmTC_c4e7a.dir/CheckIncludeFiles.c.obj   -c /home/karti/lmms/build/CMakeFiles/CMakeTmp/CheckIncludeFiles.c
C:/msys64/home/karti/lmms/build/CMakeFiles/CMakeTmp/CheckIncludeFiles.c:2:21: fatal error: sys/shm.h: No such file or directory
compilation terminated.
make[1]: *** [CMakeFiles/cmTC_c4e7a.dir/build.make:67: CMakeFiles/cmTC_c4e7a.dir/CheckIncludeFiles.c.obj] Error 1
make[1]: Leaving directory '/home/karti/lmms/build/CMakeFiles/CMakeTmp'
make: *** [Makefile:126: cmTC_c4e7a/fast] Error 2

Source:
/* */
#include <sys/shm.h>


int main(void){return 0;}

Determining if files sys/soundcard.h exist failed with the following output:
Change Dir: /home/karti/lmms/build/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/make.exe" "cmTC_272b9/fast"
/usr/bin/make -f CMakeFiles/cmTC_272b9.dir/build.make CMakeFiles/cmTC_272b9.dir/build
make[1]: Entering directory '/home/karti/lmms/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_272b9.dir/CheckIncludeFiles.c.obj
/mingw64/bin/gcc   @CMakeFiles/cmTC_272b9.dir/includes_C.rsp  -o CMakeFiles/cmTC_272b9.dir/CheckIncludeFiles.c.obj   -c /home/karti/lmms/build/CMakeFiles/CMakeTmp/CheckIncludeFiles.c
C:/msys64/home/karti/lmms/build/CMakeFiles/CMakeTmp/CheckIncludeFiles.c:2:27: fatal error: sys/soundcard.h: No such file or directory
compilation terminated.
make[1]: *** [CMakeFiles/cmTC_272b9.dir/build.make:67: CMakeFiles/cmTC_272b9.dir/CheckIncludeFiles.c.obj] Error 1
make[1]: Leaving directory '/home/karti/lmms/build/CMakeFiles/CMakeTmp'
make: *** [Makefile:126: cmTC_272b9/fast] Error 2

Source:
/* */
#include <sys/soundcard.h>


int main(void){return 0;}

Determining if files soundcard.h exist failed with the following output:
Change Dir: /home/karti/lmms/build/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/make.exe" "cmTC_e2122/fast"
/usr/bin/make -f CMakeFiles/cmTC_e2122.dir/build.make CMakeFiles/cmTC_e2122.dir/build
make[1]: Entering directory '/home/karti/lmms/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_e2122.dir/CheckIncludeFiles.c.obj
/mingw64/bin/gcc   @CMakeFiles/cmTC_e2122.dir/includes_C.rsp  -o CMakeFiles/cmTC_e2122.dir/CheckIncludeFiles.c.obj   -c /home/karti/lmms/build/CMakeFiles/CMakeTmp/CheckIncludeFiles.c
C:/msys64/home/karti/lmms/build/CMakeFiles/CMakeTmp/CheckIncludeFiles.c:2:23: fatal error: soundcard.h: No such file or directory
compilation terminated.
make[1]: *** [CMakeFiles/cmTC_e2122.dir/build.make:67: CMakeFiles/cmTC_e2122.dir/CheckIncludeFiles.c.obj] Error 1
make[1]: Leaving directory '/home/karti/lmms/build/CMakeFiles/CMakeTmp'
make: *** [Makefile:126: cmTC_e2122/fast] Error 2

Source:
/* */
#include <soundcard.h>


int main(void){return 0;}

Determining if files sys/ioctl.h exist failed with the following output:
Change Dir: /home/karti/lmms/build/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/make.exe" "cmTC_6df38/fast"
/usr/bin/make -f CMakeFiles/cmTC_6df38.dir/build.make CMakeFiles/cmTC_6df38.dir/build
make[1]: Entering directory '/home/karti/lmms/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_6df38.dir/CheckIncludeFiles.c.obj
/mingw64/bin/gcc   @CMakeFiles/cmTC_6df38.dir/includes_C.rsp  -o CMakeFiles/cmTC_6df38.dir/CheckIncludeFiles.c.obj   -c /home/karti/lmms/build/CMakeFiles/CMakeTmp/CheckIncludeFiles.c
C:/msys64/home/karti/lmms/build/CMakeFiles/CMakeTmp/CheckIncludeFiles.c:2:23: fatal error: sys/ioctl.h: No such file or directory
compilation terminated.
```

**Comment by jasp00** [comment]
I do not understand the title. Where is the check for sys/time.h for sndfile? According to the configure log, the result is `-- Looking for include file sys/time.h - found`. Is this issue "configure with MSYS does not work out of the box"? How is this environment with MSYS installed? I do not see anything like `mingw32-qt` in the `mingw-get` installer.

**Comment by knockshore** [comment]
Sorry for not being clear. Wrong configure log, replaced it and I have also added the CMake Error log in issue description.

I followed this article for installation on Windows: https://github.com/LMMS/lmms/wiki/Compiling-lmms-(On-Windows). 
I'll debug the CMakeLists.txt to see if I can find and fix anything.

**Comment by jasp00** [comment]
MSYS2 does not work with Wine. A Windows environment is needed or perhaps [Wine Staging](https://wine-staging.com/).

**Comment by tresf** [comment]
Hmm... libsoundio should be disabled since https://github.com/LMMS/lmms/commit/ac59d794fb41ee6c9dafc3c517db8c4c41d44211 but perhaps it is being ignored.

From the pictures/logs I assume you're launching from [MSYS2](https://msys2.github.io/)?  (not msys, but rather msys2).  It's been a while since anyone has attempted that tutorial, so if you find a fix, please let us know.  It will be a while before I'm able to create a msys2 build environment to reproduce.

I also haven' tried since Windows 10 added experimental [bash/apt-get support](http://www.howtogeek.com/249966/how-to-install-and-use-the-linux-bash-shell-on-windows-10/) so not sure if that could cause a conflict.

**Comment by knockshore** [comment]
Yes. MSYS2. Sure. I will try to fix it.

**Comment by Umcaruje** [comment]
@knockshore can you check if this was fixed with the merge #3369, cause I don't have any errors when compiling with this patch.

**Comment by zonkmachine** [comment]
No response, closing!
@knockshore Please reopen if you still see this on a later build, lmms-1.2.0-rc5 or later.
